pipeline_api/src/main/java/com/splicemachine/pipeline/config/SharedWriteConfiguration.java

```diff
@@ -105,7 +105,7 @@ public WriteResponse partialFailure(BulkWriteResult result, BulkWrite request) t
                         context.failed(main, mutationResult);
                     }
                 }
-                return WriteResponse.IGNORE;
+                return WriteResponse.THROW_ERROR;
             }
```

Comment:
Changing the `WriteResponse` to `THROW_ERROR` will cause the caller is throw an exception, in `BuldWriteAction`:
```
WriteResponse writeResponse=writeConfiguration.partialFailure(bulkWriteResult,currentBulkWrite);
                        switch(writeResponse){
                            case THROW_ERROR:
                                partialThrownErrorRows.add(currentBulkWrite.getSize());
                                thrown=true;
                                throw parseIntoException(bulkWriteResult);
```
The thrown exception informs the caller of the `BulkWriteAction` (which in our case, could be yet another `WriteHandler`) that something went wrong so we also cancel persisting the mutations during the call to `flush`.

Without changing this, I was running into the following problem that is best illustrated with an example; we have three tables `parent`, `child`, and `grandChild` related to each other with FK relationships (notice that `child`'s FK has `on delete cascade` while `grandChild`'s FK has `on delete no action`):
```
create table parent(col1 int, col2 varchar(2), col3 int, col4 int, primary key (col2, col4));
insert into parent values (1, 'a', 1, 1);
insert into parent values (2, 'b', 2, 2);
insert into parent values (3, 'c', 3, 3);
create table child(col1 int primary key, col2 varchar(2), col3 int, col4 timestamp, constraint fkey_1 foreign key(col2, col3) references parent(col2, col4) on delete cascade);
insert into child values (1, 'b', 2, '2020-09-09 10:10:10.123456');
insert into child values (400, 'a', 1, '2019-09-09 10:10:10.123456');
insert into child values (2, 'b', 2, '2018-09-09 10:10:10.123456');
create table grandChild(col1 int references child(col1) on delete no action, col2 varchar(2));
insert into grandChild values (1, 'z');
insert into grandChild values (1, 'z');
insert into grandChild values (1, 'z');
insert into grandChild values (400, 'y');
```
If we attempt to delete from parent:
```
delete from parent where col1 = 2;
```
the deletion will remove the row `(2, 'b', 2, 2)` from `parent` and also cause these rows to be deleted from `child`:
```
(1, 'b', 2, '2020-09-09 10:10:10.123456')
(2, 'b', 2, '2018-09-09 10:10:10.123456')
```
However the deletion of `(1, 'b', 2, '2020-09-09 10:10:10.123456')` will violate the `grandChild`'s FK since it is referenced by the following rows:
```
(1, 'z')
(1, 'z')
(1, 'z')
```
With `WriteResponse.IGNORE;`, the failure triggered by the child foreign key interceptor is not send to the parent foreign key interceptor, this means the deletion on the `parent` works but fails on the `child`. As a result, we get the following inconsistent results:
```
splice> select * From parent;
COL1       |C&|COL3       |COL4
--------------------------------------
1          |a |1          |1
3          |c |3          |3
2 rows selected
ELAPSED TIME = 29 milliseconds
splice> select * from child;
COL1       |COL2|COL3       |COL4
----------------------------------------------------------
1          |b   |2          |2020-09-09 10:10:10.123456
2          |b   |2          |2018-09-09 10:10:10.123456
400        |a   |1          |2019-09-09 10:10:10.123456
3 rows selected
ELAPSED TIME = 14 milliseconds
splice> select * from grandchild;
COL1       |COL2
----------------
1          |z
1          |z
1          |z
400        |y
```
With `WriteResponse.THROW_ERROR` the parent foreign key interceptor gets the notification and also cancels the deletion causing the overall operation to rollback maintaining referential integrity of the database.

I honestly still do not know the *full* scope of changing the `WriteRespose` like that, but it looks to be more correct than the previous behavior (to simply ignore partial failures), I think also @dgomezferro agrees on this. I am hoping we have enough test coverage that can unravel any potential issue introduced with this.